source/common/filesystem/file_shared_impl.cc

```diff
@@ -9,12 +9,12 @@ Api::IoError::IoErrorCode IoFileError::getErrorCode() const { return IoErrorCode
 
 std::string IoFileError::getErrorDetails() const { return ::strerror(errno_); }
 
-Api::IoCallBoolResult FileSharedImpl::open() {
+Api::IoCallBoolResult FileSharedImpl::open(const FlagSet in) {
   if (isOpen()) {
     return resultSuccess<bool>(true);
   }
 
-  openFile();
+  openFile(translateFlag(in));
```

Comment:
I would do the translation inside the platform specific functions. I don't think it needs to be done at this layer.